apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/script-based-flow.tsx

```diff
@@ -492,6 +598,113 @@ export const ScriptBasedFlow: FunctionComponent<AdaptiveScriptsPropsInterface> =
         />
     );
 
+    /**
+     * Render the secret list dropdown.
+     */
+    const renderSecretListDropdown = (): ReactElement => {
+        return(
+            <Dropdown
+                defaultOpen={ !isSecretListLoading }
+                labeled
+                button
+                upward={ false }
+                options={ filteredSecretList }
+                icon ={
+                    <GenericIcon
+                        hoverable
+                        transparent
+                        defaultIcon
+                        size="micro"
+                        hoverType="rounded"
+                        icon={ getOperationIcons().keyIcon }
+                        data-testid={
+                            `${ testId }-code-editor-secret-selection`
+                        }
+                    />
+                }
+            >
+                <Dropdown.Menu>
+                    <Input
+                        data-testid={ `${ testId }-secret-search` }
+                        icon="search"
+                        iconPosition="left"
+                        className="search"
+                        placeholder= { t("console:develop.features.applications.edit.sections.signOnMethod" +
+                                        ".sections.authenticationFlow.sections.scriptBased.secretsList.search") }
+                        onChange={ ( data:ChangeEvent<HTMLInputElement>
+                        ) => {
+                            if(!data.currentTarget.value) {
+                                setFilteredSecretList(secretList);
+                            } else {
+                                setFilteredSecretList(secretList.
+                                filter(x => x.secretName.includes(
```

Comment:
Better to add a meaningful name for the predicate.
```suggestion
                                filter(secret: <TYPE> => secret.secretName.includes(
```

Reply:
Fixed